### PR TITLE
updated git metadata docs page based on akit docs

### PR DESCRIPTION
**Pull request by Lewis-Seiden**

### CODE REVIEW

File: source/docs/software/advanced-gradlerio/deploy-git-data.rst [hunk 1]
```diff
@@ -1,113 +1,36 @@
 Including Git Data in Deploy
 ============================
 
-This article will go over how to include information from Git, such as branch name or commit hash, into the robot code. This is necessary for using such information in robot code, such as printing out commit hash and branch name when the robot starts.
+This article will go over how to include information from Git, such as branch name or commit hash, into the robot code using the `gversion <https://github.com/lessthanoptimal/gversion-plugin>`__ gradle plugin. This is necessary for using such information in robot code, such as printing out commit hash and branch name when the robot starts.
 
-.. note:: Git must be in the path for this to work. This should be enabled by default when `installing Git <https://git-scm.com/downloads>`__.
+- Installing gversion
```
Comment (rzblue) [inline source/docs/software/advanced-gradlerio/deploy-git-data.rst:6]: This should be a heading

File: source/docs/software/advanced-gradlerio/deploy-git-data.rst [hunk 2]
```diff
@@ -1,113 +1,36 @@
...
-This example uses `git rev-parse <https://git-scm.com/docs/git-rev-parse>`__ to extract data the name of the current branch. The Git command used for this is:
-
-.. code-block:: console
-
-   $ git rev-parse --abbrev-ref HEAD
-
-The ``--abbrev-ref`` flag tells Git to use a short version of the name for the current commit that ``rev-parse`` is acting on. When HEAD is the most recent commit on a branch, this will return the name of that branch.
-
-Next, create a new task in the ``build.gradle`` file that will run the above Git command and write it to a file in the ``src/main/deploy`` directory. For example, the following is an example task named ``writeBranchName`` that will write the branch name to a file named ``branch.txt``.
+To install gversion add the following line to the plugins block of `build.gradle`.
```
Comment (rzblue) [inline source/docs/software/advanced-gradlerio/deploy-git-data.rst:8]:
```suggestion
To install gversion add the following line to the plugins block of ``build.gradle``.
```
Inline code blocks use 2 backticks

File: source/docs/software/advanced-gradlerio/deploy-git-data.rst [hunk 3]
```diff
@@ -1,113 +1,36 @@
...
-   }
+  plugins {
+      // ...
+      id "com.peterabeles.gversion" version "1.10"
+  }
 
-This registers a `Gradle task <https://docs.gradle.org/current/userguide/tutorial_using_tasks.html>`__ that uses the above Git command, saves the output to a variable, and then writes it to a file. Since it was written to the ``src/main/deploy`` directory, it will be included in the jar file deployed to the robot and accessible in code.
-
-The next step is to make the deploy task depend on the task you created, so that it will automatically run before the code is deployed. This example uses the task name ``writeBranchName`` from the previous example, but it should be replaced with the name of the task in your ``build.gradle``.
+Then add the following block to `build.gradle` to add `createVersionFile` as a dependancy of `compileJava`. The `timeZone` field can be modified based on your team's time zone based on `this list of timezone IDs <https://docs.oracle.com/middleware/12211/wcs/tag-ref/MISC/TimeZones.html>`__.
```
Comment (rzblue) [inline source/docs/software/advanced-gradlerio/deploy-git-data.rst:17]: Explain what this does- similar to the previous contents that said "This registers a Gradle task that uses the above Git command...."

File: source/docs/software/advanced-gradlerio/deploy-git-data.rst [hunk 4]
```diff
@@ -1,113 +1,36 @@
...
-Ignoring Generated Files with Git
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-Since these files include data that is already tracked by Git and are regenerated every time code is deployed, it is recommended to not track these changes with Git by using the `gitignore <https://git-scm.com/docs/gitignore>`__ file. This file should exist by default in any project generated by the WPILib VS Code extension. Below is an example that continues to use the ``branch.txt`` and ``commit.txt`` file names:
-
-.. code-block::
-
-  src/main/deploy/branch.txt
-  src/main/deploy/commit.txt
-  ...
```
Comment (rzblue) [inline source/docs/software/advanced-gradlerio/deploy-git-data.rst:98]: This should stay with changes to make it make sense

**Review by rzblue — CHANGES_REQUESTED** [review]
Needs some formatting changes and clarifications

This should also include an example of the generated class and how to use it

File: source/docs/software/advanced-gradlerio/deploy-git-data.rst [hunk 5]
```diff
@@ -28,7 +29,34 @@ Then add the following block to `build.gradle` to add `createVersionFile` as a d
     indent       = "  "
   }
 
-You should also add the `BuildConstants.java` file to the repository `.gitignore`:
+To test this run a build of your project by clicking the WPILib logo in the top right of VS Code and searching for "Build Robot Code". Once the build has finished running there should be a file called ``BuildConstants.java`` in your Java folder. The following is an example of what this file should look like:
```
Comment (Daltz333) [inline source/docs/software/advanced-gradlerio/deploy-git-data.rst:32]: There is a run-on sentence and the sentence doesn't read well. Can you rephrase it?

File: source/docs/software/advanced-gradlerio/deploy-git-data.rst [hunk 6]
```diff
@@ -28,7 +29,34 @@ Then add the following block to `build.gradle` to add `createVersionFile` as a d
...
+    public static final long BUILD_UNIX_TIME = 1698424197122L;
+    public static final int DIRTY = 0;
+
+    private BuildConstants(){}
+  }
+
+Ignoring Generated Files with Git
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Since these files include data that is already tracked by Git and are regenerated every time code is deployed, it is recommended to not track these changes with Git by using the `gitignore <https://git-scm.com/docs/gitignore>`__ file. This file should exist by default in any project generated by the WPILib VS Code extension. Below is the line you should add to ignore the generated file:
```
Comment (Daltz333) [inline source/docs/software/advanced-gradlerio/deploy-git-data.rst:59]: The language here is pretty stiff.

File: source/docs/software/advanced-gradlerio/deploy-git-data.rst [hunk 7]
```diff
@@ -14,7 +15,7 @@ To install gversion add the following line to the plugins block of `build.gradle
       id "com.peterabeles.gversion" version "1.10"
   }
 
-Then add the following block to `build.gradle` to add `createVersionFile` as a dependancy of `compileJava`. The `timeZone` field can be modified based on your team's time zone based on `this list of timezone IDs <https://docs.oracle.com/middleware/12211/wcs/tag-ref/MISC/TimeZones.html>`__.
+Next we need to tell gradle to make the file containing our git data when we build the project. To do that add the following block to ``build.gradle``, which adds ``createVersionFile`` as a dependency of ``compileJava``. The ``timeZone`` field can be modified based on your team's time zone based on `this list of timezone IDs <https://docs.oracle.com/middleware/12211/wcs/tag-ref/MISC/TimeZones.html>`__. The other parameters tell gversion where to put our file, what Java package it should be in, the name of the file, how to show the commit date, and indentation formatting.
```
Comment (Daltz333) [inline source/docs/software/advanced-gradlerio/deploy-git-data.rst:18]: We should avoid "we".

**Review by Daltz333 — CHANGES_REQUESTED** [review]

**Comment by sciencewhiz** [comment]
Does this address #2182?

File: source/docs/software/advanced-gradlerio/deploy-git-data.rst [hunk 8]
```diff
@@ -1,113 +1,68 @@
...
-This example uses `git rev-parse <https://git-scm.com/docs/git-rev-parse>`__ to extract data the name of the current branch. The Git command used for this is:
-
-.. code-block:: console
-
-   $ git rev-parse --abbrev-ref HEAD
-
-The ``--abbrev-ref`` flag tells Git to use a short version of the name for the current commit that ``rev-parse`` is acting on. When HEAD is the most recent commit on a branch, this will return the name of that branch.
-
-Next, create a new task in the ``build.gradle`` file that will run the above Git command and write it to a file in the ``src/main/deploy`` directory. For example, the following is an example task named ``writeBranchName`` that will write the branch name to a file named ``branch.txt``.
+To install gversion add the following line to the plugins block of ``build.gradle``. This tells Gradle to use gversion in the project.
```
Comment (Daltz333) [inline source/docs/software/advanced-gradlerio/deploy-git-data.rst:9]: Punctuation needs to be added here

File: source/docs/software/advanced-gradlerio/deploy-git-data.rst [hunk 9]
```diff
@@ -1,113 +1,68 @@
...
-         projectDir.toString() + "/src/main/deploy",
-         // File name to write to
-         "branch.txt"
-      ).text = branch // Set the contents of the file to the variable branch
-   }
-
-This registers a `Gradle task <https://docs.gradle.org/current/userguide/tutorial_using_tasks.html>`__ that uses the above Git command, saves the output to a variable, and then writes it to a file. Since it was written to the ``src/main/deploy`` directory, it will be included in the jar file deployed to the robot and accessible in code.
-
-The next step is to make the deploy task depend on the task you created, so that it will automatically run before the code is deployed. This example uses the task name ``writeBranchName`` from the previous example, but it should be replaced with the name of the task in your ``build.gradle``.
+In order to generate the file when building the project, add the following block to the bottom or otherwise at the root of ``build.gradle``.
```
Comment (Daltz333) [inline source/docs/software/advanced-gradlerio/deploy-git-data.rst:19]: I think the latter fragment "otherwise..." Is unnecessary

File: source/docs/software/advanced-gradlerio/deploy-git-data.rst [hunk 10]
```diff
@@ -1,113 +1,68 @@
...
+    dateFormat   = "yyyy-MM-dd HH:mm:ss z"
+    timeZone     = "America/New_York" // Use preferred time zone
+    indent       = "  "
+  }
 
-Similar to the previous example, ``git rev-parse`` will be used to parse the current commit hash. The Git command used for this is:
+The ``srcDir``, ``classPackage``, and ``className`` parameters tell the plugin where to put the manifest file, and what to name it. The ``timeZone`` field can be set to your team's time zone based on `this list of timezone IDs <https://docs.oracle.com/middleware/12211/wcs/tag-ref/MISC/TimeZones.html>`__. The ``dateFormat`` parameter is based on `this Java class <https://docs.oracle.com/javase/8/docs/api/java/text/SimpleDateFormat.html>`__.
 
-.. code-block:: console
+To test this run a build of your project by clicking the WPILib logo in the top right of VS Code and selecting the "Build Robot Code" option. Once the code has finished building, there should be a file called ``BuildConstants.java`` in your ``src/main/java`` folder. The following is an example of what this file should look like:
```
Comment (Daltz333) [inline source/docs/software/advanced-gradlerio/deploy-git-data.rst:35]: Run on sentence here

**Review by Daltz333 — CHANGES_REQUESTED** [review]

File: source/docs/software/advanced-gradlerio/deploy-git-data.rst (hunk 8 above)
Comment (Lewis-Seiden) [reply source/docs/software/advanced-gradlerio/deploy-git-data.rst:9]: what punctuation is missing?

**Review by Starlight220 — COMMENTED** [review]
How does this behave if the project isn't stored in git?

**Comment by Lewis-Seiden** [comment]
> How does this behave if the project isn't stored in git?

it generates `BuildConstants.java` with most fields set to a "not found" or default value

**Comment by jasondaming** [comment]
It looks good enough to me but I am not the grammar....expert.

**Review by Daltz333 — APPROVED** [review]
